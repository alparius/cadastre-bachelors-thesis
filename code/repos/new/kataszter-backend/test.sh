#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
