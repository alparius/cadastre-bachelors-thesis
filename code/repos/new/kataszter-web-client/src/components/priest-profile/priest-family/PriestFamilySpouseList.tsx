import React from "react";
import { Card } from "semantic-ui-react";

import IPriestProfileData from "../../../data/priest/PriestProfileData";
import "../../../static/priestProfile.css";
import ItalicFM from "../../../util/styledMessages/ItalicFM";
import SpouseCard from "./SpouseCard";

interface IPriestSpouseListProps {
    priest: IPriestProfileData;
    editMode: boolean;
    setPriest(newPriest: IPriestProfileData): void;
}

const PriestFamilySpouseList: React.FC<IPriestSpouseListProps> = (props: IPriestSpouseListProps) => {
    const { priest, editMode, setPriest } = props;

    const handleDelete = (index: number) => {
        priest.spouses.splice(index, 1);
        setPriest({
            ...priest,
            spouses: priest.spouses
        });
    };

    const handleChangeOnSpouse = (labelName: string, value: any, index: number) => {
        setPriest({
            ...priest,
            spouses: priest.spouses.map((spouse, i) => (i === index ? { ...spouse, [labelName]: value } : { ...spouse }))
        });
    };

    if (!priest.spouses || priest.spouses.length === 0) {
        return <ItalicFM id="general.noData" />;
    } else {
        return (
            <Card.Group stackable doubling itemsPerRow={2}>
                {priest.spouses.map((_, index: number) => (
                    <SpouseCard
                        key={index}
                        spouse={priest.spouses[index]}
                        index={index}
                        editMode={editMode}
                        handleDelete={handleDelete}
                        handleChangeOnSpouse={handleChangeOnSpouse}
                    />
                ))}
            </Card.Group>
        );
    }
};

export default PriestFamilySpouseList;
