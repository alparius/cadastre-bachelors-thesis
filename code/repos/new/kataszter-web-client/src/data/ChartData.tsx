interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
