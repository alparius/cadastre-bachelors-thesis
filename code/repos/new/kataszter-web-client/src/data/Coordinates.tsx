interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
