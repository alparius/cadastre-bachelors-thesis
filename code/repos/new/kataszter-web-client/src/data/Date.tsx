interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
