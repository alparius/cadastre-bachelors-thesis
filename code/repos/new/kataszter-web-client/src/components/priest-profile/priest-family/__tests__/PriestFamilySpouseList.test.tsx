/*===================
  PriestFamilySpouseList.unit.test.tsx
 ===================*/
import React from "react";

import IPriestProfileData from "../../../../data/priest/PriestProfileData";
import ISpouse from "../../../../data/priest/Spouse";
import { NewTestPriestProfileData } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../../util/test/test-wrapper";
import PriestFamilySpouseList from "../PriestFamilySpouseList";

const handlers = {
    setPriest: jest.fn()
};

const priest: IPriestProfileData = NewTestPriestProfileData;

describe("PriestFamilySpouseList element", () => {
    it("should render a family spouse list container when the spouses is not empty", () => {
        const container = createComponentWithIntlAndRouter(<PriestFamilySpouseList priest={priest} editMode={false} {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a `no data` when the spouses is empty", () => {
        const emptySpouses: ISpouse[] = [];
        priest.spouses = emptySpouses;

        const container = createComponentWithIntlAndRouter(<PriestFamilySpouseList priest={priest} editMode={false} {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });
});
