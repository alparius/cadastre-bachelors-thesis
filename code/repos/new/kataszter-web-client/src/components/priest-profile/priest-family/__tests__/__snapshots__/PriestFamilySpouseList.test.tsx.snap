// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestFamilySpouseList element should render a \`no data\` when the spouses is empty 1`] = `
<i>
  No data
</i>
`;

exports[`PriestFamilySpouseList element should render a family spouse list container when the spouses is not empty 1`] = `
<div
  className="ui doubling stackable two cards"
>
  <div
    className="ui fluid card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="content"
      >
        <div
          className="header"
          id="header"
        >
          <h2
            className="ui header"
          >
            Házastárs
          </h2>
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="ui list"
          role="list"
        >
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Date of birth
                  :
                </b>
                 
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  2000.06.06
                </span>
              </div>
            </form>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Date of marriage
                  :
                </b>
                 
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  2000.06.06
                </span>
              </div>
            </form>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Job
                  :
                </b>
                 
                Szavazati joggal rendelkező
              </div>
            </form>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Father
                  :
                </b>
                 
                Lajos
              </div>
            </form>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Mother
                  :
                </b>
                 
                Laci
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
